python/pyspark/mllib/common.py

```diff
@@ -70,8 +70,8 @@ def _py2java(sc, obj):
         obj = _to_java_object_rdd(obj)
     elif isinstance(obj, SparkContext):
         obj = obj._jsc
-    elif isinstance(obj, list) and (obj or isinstance(obj[0], JavaObject)):
-        obj = ListConverter().convert(obj, sc._gateway._gateway_client)
+    elif isinstance(obj, list):
+        obj = ListConverter().convert([_py2java(sc, x) for x in obj], sc._gateway._gateway_client)
```

Comment:
@mengxr Let me guess, when I encounter an `Array[ndarray]`, this line of code will try to translate the `ndarray` one by one. For each `ndarray`, the `else` statement will be triggered, `PikleSerializer` will serialize the `ndarray`, then the helper function in JVM will turn it back to `Vector`.